lib/public/appframework/http/streamresponse.php

```
@@ -0,0 +1,34 @@
+<?php
+/**
+ * @author Bernhard Posselt
+ * @copyright 2015 Bernhard Posselt <dev@bernhard-posselt.com>
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later.
+ * See the COPYING-README file.
+ */
+
+namespace OCP\AppFramework\Http;
+
+
+class StreamResponse extends Response implements ICallbackResponse {
+
+    private $filePath;
+
+    /**
+     * @param string $filePath the path to the file which should be streamed
+     */
+    public function __construct ($filePath) {
+        $this->filePath = $filePath;
+    }
+
+
+    /**
+     * Streams the file using readfile
+     */
+    public function callback () {
+        @readfile($this->filepath);
```

Good question, I actually don't know because there is nothing that catches the exception. ATM it will just silently fail if the content is not available which is not a good idea i think. Just throw any exception?

Ideally we'd have a 404 if the file does not exist, the question is just where and how we should throw the exception. Maybe it would be best to just leave this up to the developer, e.g.:

```php

public function controllerMethod($fileId) {
    try {
        $filePath = $this->service->findPath($fileId);
        return new StreamResponse($filePath);
    } catch (FileDoesNotExistException $e) {
        return new Response(Http::STATUS_NOT_FOUND);
    }
}
```
otherwise we'd need custom middleware that checks the filepaths of all StreamResponses and replaces it with 404 responses which is akward I think. 

As for xsendfile: a good idea but how do we handle php streams? How would we return a file that is on an FTP? It is fast but probably has a different use case because of the imposed limits and should go into a separate response that falls back to readfile if xsendfile does not exist